src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -378,14 +378,21 @@ public sealed partial class SocketsHttpHandler : System.Net.Http.HttpMessageHand
         protected internal override System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> SendAsync(System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) { throw null; }
         public bool EnableMultipleHttp2Connections { get { throw null; } set { } }
         public Func<SocketsHttpConnectionContext, System.Threading.CancellationToken, System.Threading.Tasks.ValueTask<System.IO.Stream>>? ConnectCallback { get { throw null; } set { } }
+        public Func<SocketsHttpPlaintextStreamFilterContext, System.Threading.CancellationToken, System.Threading.Tasks.ValueTask<System.IO.Stream>>? PlaintextStreamFilter { get { throw null; } set { } }
     }
     public sealed class SocketsHttpConnectionContext
     {
         internal SocketsHttpConnectionContext() { }
         public DnsEndPoint DnsEndPoint { get { throw null; } }
         public HttpRequestMessage RequestMessage { get { throw null; } }
     }
-
+    public sealed class SocketsHttpPlaintextStreamFilterContext
+    {
+        internal SocketsHttpPlaintextStreamFilterContext() { }
+        public System.IO.Stream PlaintextStream { get { throw null; } }
+        public Version HttpVersion { get { throw null; } }
+        public HttpRequestMessage RequestMessage { get { throw null; } }
```

Comment:
I asked offline, but just to make sure it's not lost, was wondering if the additional end point context from the ConnectCallback is relevant here as well.  If not, that's fine (and of course we could always add it in the future if it proved valuable).  In some cases, relevant end point data could potentially also be fished out of the PlaintextStream.

Reply:
Possibly -- but absent a clear scenario for it, I've just left it off for now.